src/discrete-layer/models/discreteLayersStore.ts

```diff
@@ -221,12 +227,18 @@ export const discreteLayersStore = ModelBase
       self.selectedLayer =  layer ? {...layer} : undefined;
     }
 
-    function setTabviewData(tabView: TabViews): void {
+    function setTabviewData(tabView: TabViews, customLayersImages?: ILayerImage[]): void {
       if (self.tabViews) {
         const idxTabViewToUpdate = self.tabViews.findIndex((tab) => tab.idx === tabView);
 
-        self.tabViews[idxTabViewToUpdate].selectedLayer = self.selectedLayer ? { ...self.selectedLayer } : undefined;
-        self.tabViews[idxTabViewToUpdate].layersImages = self.layersImages ? [ ...self.layersImages ]: [];
+        if(customLayersImages) {
+          const preparedLayersImages = getPreparedLayersImages([...customLayersImages]);
```

Comment:
function setTabviewData() called number of times with already "prepared" data think it's not "biggy", is it ok from your side?

Reply:
This is just the case where we want to apply custom layers images directly to the tab view (When we change recordType), so just in that case we need to prepare the data before the insertion. Even if we send in already prepared data it wont cause problems besides calling the method again redundantly.